java/example/calypso/pc/UseCase8_StoredValue_DebitInSession/src/main/java/org/eclipse/keyple/example/calypso/pc/usecase8/StoredValue_DebitInSession_Pcsc.java

```diff
@@ -0,0 +1,204 @@
+        poReader = CalypsoUtilities.getDefaultPoReader();
+
+        // Get a SAM reader ready to work with Calypso PO. Use the getReader helper method from the
+        // CalypsoUtilities class.
+        SeResource<CalypsoSam> samResource = CalypsoUtilities.getDefaultSamResource();
+
+        logger.info("=============== UseCase Calypso #8: Stored Value Debit in Session ==");
+        logger.info("= PO Reader  NAME = {}", poReader.getName());
+        logger.info("= SAM Reader  NAME = {}", samResource.getSeReader().getName());
+
+        if (selectPo()) {
+            // Security settings
+            // Both Reload and Debit SV logs are requested
+            PoSecuritySettings poSecuritySettings =
+                    new PoSecuritySettings.PoSecuritySettingsBuilder(samResource)//
+                            .svGetLogReadMode(SvSettings.LogRead.ALL).build();
+
+            // Create the PO resource
+            SeResource<CalypsoPo> poResource;
+            poResource = new SeResource<CalypsoPo>(poReader, calypsoPo);
+
+            PoTransaction poTransaction = new PoTransaction(poResource, poSecuritySettings);
+
+            // Read the EventLog file at the Session Opening
+            poTransaction.prepareReadRecordFile(CalypsoClassicInfo.SFI_EventLog,
+                    CalypsoClassicInfo.RECORD_NUMBER_1);
+
+            // Open a secure session (DEBIT level) and execute the prepared command
+            poTransaction
+                    .processOpening(PoTransaction.SessionSetting.AccessLevel.SESSION_LVL_DEBIT);
+
+            // Get and display the EventLog data
+            ElementaryFile efEventLog = calypsoPo.getFileBySfi(CalypsoClassicInfo.SFI_EventLog);
+
+            String eventLog = ByteArrayUtil.toHex(efEventLog.getData().getContent());
+            logger.info("File Event log: {}", eventLog);
+
+            // Prepare a SV Debit (this command could also have been placed before processOpening
+            // since it is not followed by any other command)
+            poTransaction.prepareSvGet(SvSettings.Operation.DEBIT, SvSettings.Action.DO);
+
+            // Execute the prepared command
+            poTransaction.processPoCommandsInSession();
+
+            // Display the current SV status
+            logger.info("Current SV status (SV Get for DEBIT):");
+            logger.info(". Balance = {}", calypsoPo.getSvBalance());
+            logger.info(". Last Transaction Number = {}", calypsoPo.getSvLastTNum());
+
+            // To easily display the content of the logs, we use here the toString method which
+            // exports the data in JSON format.
+            String loadLogRecordJson = prettyPrintJson(calypsoPo.getSvLoadLogRecord().toString());
+            String debitLogRecordJson =
+                    prettyPrintJson(calypsoPo.getSvDebitLogLastRecord().toString());
+            logger.info(". Load log record = {}", loadLogRecordJson);
+            logger.info(". Debit log record = {}", debitLogRecordJson);
+
```

Comment:
maybe add a comment saying that the terminal could check if the SAM module which has operated the last SV Reload is blacklisted or not.